api/rest/client/methods_test.go

```diff
@@ -224,3 +227,64 @@ func TestGetConnectGraph(t *testing.T) {
 		t.Fatal("Bad graph")
 	}
 }
+
+func TestClient_WaitFor(t *testing.T) {
+	ci, _ := cid.Decode(test.TestCid1)
+	type args struct {
+		ci        *cid.Cid
+		local     bool
+		target    api.TrackerStatus
+		checkFreq time.Duration
+	}
+	tests := []struct {
+		name string
+		args args
+		want api.TrackerStatus
+	}{
+		{
+			"unpin",
+			args{
+				ci,
+				false,
+				api.TrackerStatusUnpinned,
+				2 * time.Second,
+			},
+			api.TrackerStatusUnpinned,
+		},
+		{
+			"pin",
+			args{
+				ci,
+				false,
+				api.TrackerStatusPinned,
+				2 * time.Second,
+			},
+			api.TrackerStatusPinned,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, api := testClient(t)
+			defer api.Shutdown()
+
+			var err error
+			switch tt.name {
+			case "unpin":
+				err = c.Pin(ci, 1, 1, "test-pin")
+			case "pin":
+				err = c.Unpin(ci)
+			}
+
+			t.Errorf("failed to perform setup %s: %v", tt.name, err)
+
+			statusCh := c.WaitFor(tt.args.ci, tt.args.local, tt.args.target, tt.args.checkFreq)
+			for {
+				select {
+				case stat := <-statusCh:
+					log.Printf("%#v\n", stat)
```

Comment:
I'm guessing this is just incomplete now because you are debugging but eventually it will be good to assert tt.target == stat here.